protocol/topic.go

```diff
@@ -89,7 +89,7 @@ const (
 )
 
 var regexFiveElementTopic = regexp.MustCompile("^([^/]+)/([^/]+)/([^/]+)/([^/]+)/([^/]+)$")
-var regexSixElementTopic = regexp.MustCompile("^([^/]+)/([^/]+)/([^/]+)/([^/]+)/([^/]+)/([^/]+)$")
+var regexSixElementTopic = regexp.MustCompile("^([^/]+)/([^/]+)/([^/]+)/([^/]+)/([^/]+)/(.*)$")
```

Comment:
Can we simplify the regexp. here and replace the two separate ones with only one - something like e.g. `"^([^/]+)/([^/]+)/([^/]+)/([^/]+)/([^/]+)(/([^/]{1}.+))?$"` 
?
It's not a strict required RFC matcher but it's a start :)
That way only one regexp will be compiled and useful in both cases as well as helpful to simplify the logic using it.